src/Init/Data/List/Lemmas.lean

```diff
@@ -745,6 +745,58 @@ theorem foldr_hom (f : β₁ → β₂) (g₁ : α → β₁ → β₁) (g₂ :
     (H : ∀ x y, g₂ x (f y) = f (g₁ x y)) : l.foldr g₂ (f init) = f (l.foldr g₁ init) := by
   induction l <;> simp [*, H]
 
+/--
+Prove a proposition about the result of `List.foldl`,
+by proving it for the initial data,
+and the implication that the operation applied to any element of the list preserves the property.
+
+The motive `C` can take values in `Sort _`, so this may be used to construct data,
+as well as to prove propositions.
+-/
+def foldlRecOn {C : β → Sort _} : ∀ (l : List α) (op : β → α → β) (b : β) (_ : C b)
```

Comment:
Repeating my comment from the other thread (sorry about the `elab_as_elim` not working), isn't the convention in lean4 to use `motive` and not `C`?

Reply:
I've never been happy with `motive`, as most new users find it more confusing than a generic name, but I guess that's a change to consider separately. Done.

Reply:
Thanks!